File: editor/en/[9]Palindrome Number.java
//leetcode submit region begin(Prohibit modification and deletion)
class Solution {
    public boolean isPalindrome(int x) {
        if (x < 0 || (x % 10 == 0 && x > 0)) {
            return false;
        }

        int original = x;
        int reversed = 0;

        while (x > 0) {
            //  make sure reversed * 10 + x % 10 <= Integer.MAX_VALUE
            if (reversed > (Integer.MAX_VALUE - (x % 10)) / 10) {
                return false;
            }

            reversed = reversed * 10 + x % 10;
            x /= 10;
        }

        return reversed == original;
    }
}
//leetcode submit region end(Prohibit modification and deletion)

File: editor/en/[9]Palindrome Number.js
//leetcode submit region begin(Prohibit modification and deletion)
/**
 * @param {number} x
 * @return {boolean}
 */
var isPalindrome = function (x) {
    if (x < 0 || (x % 10 === 0 && x !== 0)) {
        return false;
    }
    let res = 0;
    let num = x;
    while (x !== 0) {
        res = res * 10 + x % 10;
        x = Math.floor(x / 10);
    }
    return res === num;
};
//leetcode submit region end(Prohibit modification and deletion)
